arcgis-ios-sdk-samples/Maps/Generate offline map (overrides)/OfflineMapParameterOverridesViewController.swift

```diff
@@ -0,0 +1,274 @@
+    /// i.e. 23 has the most detail, but each tile covers a tiny area.
+    @IBOutlet weak var maxScaleLevelSlider: UISlider!
+    
+    /// The extra padding added to the extent rect to fetch a larger area, in meters.
+    @IBOutlet weak var basemapExtentBufferSlider: UISlider!
+    
+    /// Switch indicating if the system valves layer should be included in the download.
+    @IBOutlet weak var includeSystemValvesSwitch: UISwitch!
+    ///Switch indicating if the service connections layer should be included in the download.
+    @IBOutlet weak var includeServiceConnectionsSwitch: UISwitch!
+    
+    /// The minimum flow rate by which to filter features in the Hydrants layer, in gallons per minute.
+    @IBOutlet weak var minHydrantFlowRateSlider: UISlider!
+    
+    ///Switch indicating if the pipe layers should be restricted to the extent frame.
+    @IBOutlet weak var cropWaterPipesToExtentSwitch: UISwitch!
+    
+    @IBOutlet weak var minScaleLevelLabel: UILabel!
+    @IBOutlet weak var maxScaleLevelLabel: UILabel!
+    @IBOutlet weak var extentBufferLabel: UILabel!
+    @IBOutlet weak var minHydrantFlowRateLabel: UILabel!
+    
+    //MARK: - Actions
+    
+    @IBAction func sliderChangeAction(_ sender: UISlider) {
+        if sender == minScaleLevelSlider {
+            // Disallow a min value greater than the maximum
+            if sender.value > maxScaleLevelSlider.value {
+                sender.value = maxScaleLevelSlider.value
+            }
+        }
+        else if sender == maxScaleLevelSlider {
+            // Disallow a max value less than the minimum
+            if sender.value < minScaleLevelSlider.value {
+                sender.value = minScaleLevelSlider.value
+            }
+        }
+        // Update this slider's text field to display the new value
+        updateTextField(for: sender)
+    }
+    
+    //MARK: - Text field updating
+    
+    private let numberFormatter = NumberFormatter()
+    
+    private func updateTextField(for slider: UISlider){
+        if let text = numberFormatter.string(from: slider.value as NSNumber) {
+            if slider == minScaleLevelSlider {
+                minScaleLevelLabel.text = text
+            }
+            else if slider == maxScaleLevelSlider {
+                maxScaleLevelLabel.text = text
+            }
+            else if slider == basemapExtentBufferSlider {
+                extentBufferLabel.text = text
+            }
+            else if slider == minHydrantFlowRateSlider {
+                minHydrantFlowRateLabel.text = text
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Update each text field to display its slider's default value
+        for slider in [minScaleLevelSlider,
+                       maxScaleLevelSlider,
+                       basemapExtentBufferSlider,
+                       minHydrantFlowRateSlider] {
+            updateTextField(for: slider!)
```

Comment:
I'm torn over this. On one hand using a `for` loop is nice because you don't have to write the function call multiple times. On the other hand, it means doing an unnecessary comparison test for each slider. You already know which label corresponds to which slider, so if it were me, I would just update the label's value directly. I suppose you could have an `updateLabel(from slider: Slider)` function, but it would only be a single line.

Reply:
What about something like this? This is easy to read and reduces doing unnecessary comparison for each slider.

```swift
override func viewDidLoad() {
    super.viewDidLoad()
    
    typealias Parameters = (slider: UISlider, label: UILabel)
    
    // Update each text field to display its slider's default value
    let parameters: [Parameters] = [(minScaleLevelSlider, minScaleLevelLabel),
                                    (maxScaleLevelSlider, maxScaleLevelLabel),
                                    (basemapExtentBufferSlider, extentBufferLabel),
                                    (minHydrantFlowRateSlider, minHydrantFlowRateLabel)]
    
    for parameter in parameters {
        parameter.label.text = numberFormatter.string(from: parameter.slider.value as NSNumber)
    }
}
```

Reply:
You also could build a dictionary: `[UISlider: UILabel]` and reduce the overhead found in: `private func updateTextField(for slider: UISlider)`. Just some suggestions!